subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/ConnectorServices.java

```diff
@@ -57,6 +58,9 @@ public static void reset() {
 
     private static void checkJavaVersion() {
         UnsupportedJavaRuntimeException.assertUsingVersion("Gradle Tooling API", JavaVersion.VERSION_1_7);
+        if (!JavaVersion.current().isJava8Compatible()) {
+            DeprecationLogger.nagUserWith("Support for running Gradle using Java 7 has been deprecated and is scheduled to be removed in Gradle 5.0");
+        }
```

Comment:
:o: Maybe we can extend `assertUsingVersion()` such that it keeps care of calling `nagUserWith` instead of adding the individual calls to `nagUserWith`? Something like `assertUsingVersion("Gradle Tooling API", JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_8)`, where we fail if VERSION_1_7 is not fulfilled and warn if VERSION_1_8 is not fulfilled.

Reply:
Inspired by you, I put the `nagUserWith` in `checkJavaVersion` method so that each time `checkJavaVersion` is called, the java 7 deprecation warning would be called by the way.